### DISCUSSION TITLE
DetailsList and GroupedList in Fluent UI React Components v9

### DISCUSSION THREAD

**Discussion by kroudams** [microsoft/fluentui#24751] [Q&A]
Are you planning on implementing [DetailsList](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist) and [GroupedList](https://developer.microsoft.com/en-us/fluentui#/controls/web/groupedlist) components in Fluent UI React Components v9?

As on [Fluent UI React v9 Component Roadmap](https://github.com/microsoft/fluentui/wiki/Fluent-UI-React-v9-Component-Roadmap) page, I haven't seen those (there are List and Tree though. Are they going to replace DetailsList and GroupedList?).

![image](https://user-images.githubusercontent.com/87769044/189595105-a4c6dff0-a522-46f0-bc47-475e3829bd70.png)

**Comment by AkiraVoid** [comment]
Maybe you could take a look of Table. It's available now on doc site, though it's still unstable.

**Reply by kroudams** [reply]
Thank you, for your reply and suggestion.

I need to know regarding DetailsList and GroupedList components before I consider starting my current component based on one of those or ditching them and starting implementation based on Fluent UI V9 components. 
Or, as another option, maybe I just implement it myself from the scratch. 
Thus my question.

**Reply by AkiraVoid** [reply]
I don't think v9 will have DetailList or GroupedList as same as those in v8. As I know, Table component in v9 is a better alternate of DetailList, all functions provided by DetailList and GroupedList can be easily implemented by using Table.

**Reply by kroudams** [reply]
> ... all functions provided by DetailList and GroupedList can be easily implemented by using Table.

Well as far as I can see, almost all functions. 
What I need is a grouping. 
Judging by the current implementation (in V9) it is not possible with the Table.
What I need is a sort of a list with a grouping capability (more like tree view pattern)

**Reply by AkiraVoid** [reply]
The Tree and List are currently TBD, which means they even have not built up the folders... You probably need to open an issue as a feature request. Maybe you could use Accordion and TableRow (and other child components of Table) to achieve your needs.

**Reply by kroudams** [reply]
Thanks. 
Have opened a new [issue](https://github.com/microsoft/fluentui/issues/24774)

**Comment by ling1726** [comment]
Hi folks, thanks for using the discussion feature. We've currently been working on Table primitives so that users can meet design requirements. We have a headless table implementation in research and is kind of tracked under #24226. Now that I know grouping is an important feature, I will explore the feasibility. 

However, at the end of the day, I think that the accessibilty is a bigger issue. I don't see any other libraries with this kind of collapsible table feature. If you know any I'd be happy to check it out.

Popular libraries like react-table https://tanstack.com/table/v8/docs/examples/react/grouping might handle grouping but they are a headless library which does have any accessibility requirements on them. Logically for me the solution would be a tree or treegrid like component.

Perhaps our a11y start @smhigley  might give us some of her feedback on row grouping in tables ?

**Reply by kroudams** [reply]
Thank you for your answer.

Yes, I agree that the Tree component would suit better in this scenario. 
The question is, when do you think it might be implemented if at all?

By the by, if you're gonna implement it I assume it will include grouping by default.

I need to know whether you're gonna implement the Tree pattern soon (say by the end of the year) or if I will need to build it myself for my needs. 

@ling1726, @smhigley, @paulgildea, @JustSlone
